W-18564545 intro mb (#850)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/flex-agent-secure.adoc

endif::[]
:imagesdir: ../assets/images

Flex Gateway supports both the Model Context Protocol (MCP) and the Agent2Agent (A2A) Protocol, enabling you to protect your agents and systems. Flex Gateway provides centralized oversight, agent visibility, logging, and valuable insights, ensuring effective governance and security for your agent-based architecture. Flex Gateway secures agent interactions by enforcing policies across agent connections and monitoring agent activities for enhanced visibility.

== Flex Gateway A2A and MCP Support

> Flex Gateway support for these protocols enables you to secure and manage agent interactions.

Reviewer: ```suggestion
Flex Gateway enables you to secure and manage MCP and A2A agent interactions.
```

A2A::
  An open standard that defines how agents interact with each other, enabling interoperability for agents built by different teams, using different technologies, and hosted by different organizations.

MCP::
  An open protocol that defines how agents interact with tools or external context, enabling agents to connect with the data and tools they need.

== Securing Agent Interactions with Policies

> Flex Gateway includes A2A and MCP specific policies, which provide enhanced security and control over agent interactions. Flex Gateway A2A policies enable you to protect agent endpoints by rewriting Agent Cards URLs, logging or blocking sensitive information in messages, modifying prompt behavior, and with Server-Sent Events (SSE) content logging for audit and compliance needs. Flex Gateway MCP policies enable you to use attribute-based access control to manage access to MCP servers.

Reviewer: ```suggestion
Flex Gateway includes A2A and MCP specific policies that provide enhanced security and control over agent interactions. Flex Gateway A2A policies enable you to protect agent endpoints by rewriting Agent Cards URLs, logging or blocking sensitive information in messages, modifying prompt behavior, and  logging Server-Sent Events (SSE) content for compliance audits.  Flex Gateway MCP policies enable you to use attribute-based access control to manage access to MCP servers.
```

> You can also use other Flex Gateway policies to secure and manage MCP and A2A connections. Use policies such as xref:policies-included-rate-limiting.adoc[] or xref:policies-included-spike-control.adoc[] to manage the number of requests to MCP and A2A servers. Use the policies-included-message-logging.adoc[] to monitor agent interactions.

Reviewer: ```suggestion
Flex Gateway MCP and A2A server instances also support Flex Gateway's other policies. For example, use policies such as xref:policies-included-rate-limiting.adoc[] or xref:policies-included-spike-control.adoc[] to manage the number of requests to MCP and A2A servers or the policies-included-message-logging.adoc[] to monitor agent interactions.
```
